tests/test_security_path_traversal.py

```
@@ -204,3 +204,64 @@ def test_run_tailer_path_validation(tmp_path):
+            assert exc_info.value.status_code == 400
+        asyncio.run(run_tests())
+
+def test_wisdom_api_path_traversal_integration():
+    """Test wisdom API against path traversal."""
+    import asyncio
+    from fastapi import HTTPException
+    from swarm.api.routes import wisdom
+    from unittest.mock import patch, MagicMock
+    from pathlib import Path
+
+    with patch("swarm.api.routes.wisdom._get_runs_root", return_value=Path("/tmp/runs")), \
```

Ruff (0.15.4)

[error] Probable insecure usage of temporary file or directory: "/tmp/runs"

(S108)